exercises/Phil_Colebrooke/ex01_basics/src/main.cpp

```diff
@@ -1,135 +1,167 @@
+  std::string outputFilename;
+  int minWordLength;
+  std::map<std::string, int> wordsAndCounts;
+  std::vector<std::pair<int, std::string>> sorted;
+  bool success{false};
+
+  // checks that the user has provided the correct number of arguments
+  if (argc != CORRECT_NUMBER_ARGUMENTS) {
+    std::cerr
+        << "Wrong number of arguments provided. "
+           "Provide two filenames (input and output) and minimum word length."
+        << "\n";
+    return 1;
+  }
+
+  inputFilename = argv[1];
+  outputFilename = argv[2];
+
+  // the inputted min word length is converted to int
+  minWordLength = atoi(argv[3]);
+
+  success = countWords(inputFilename, wordsAndCounts, minWordLength);
+
+  if (success) {
+    sorted = orderByUsage(wordsAndCounts);
+    writeToFile(sorted, outputFilename);
 
     return 0;
-}
-
-// opens file with the provided filename
-// populates a map with each word in the file
-// and the number of occurences
-void countWords(std::string filename,
-    std::map<std::string, int>& wordCounter) {
+  }
 
-    std::ifstream file{ filename };
+  return 1;
+}
 
-    // checks the file has been opened
-    if (!file.is_open()) {
-        std::cerr << "Unable to open file.";
-        return;
+/** Opens a text file and populates a map with every unique word in the file and
+ * the number of times the word occurs
+ *
+ * @param filename The name of the input file to read
+ * @param wordsAndCounts The map to be populated
+ * @param minWordLength The minimum length of words to be added to the map
+ * @return A boolean for whether or not the map was successfully populated
+ *
+ */
+bool countWords(const std::string &inputFilename,
+                std::map<std::string, int> &wordsAndCounts,
+                const int &minWordLength) {
+
+  std::ifstream file{inputFilename};
+
+  // checks the file has been opened
+  if (!file.is_open()) {
+    std::cerr << "Unable to open input file for reading. Check that the file "
+                 "exists and its name is spelt correctly."
+              << "/n";
+    return false;
+  }
+
+  // goes through each word in the file
+  std::string word;
+  while (file >> word) {
+
+    word = cleanUpWord(word);
+
+    // only words of length at least the inputted minimum are added to the map
+    if (word.length() >= minWordLength) {
+      ++wordsAndCounts[word];
     }
+  }
 
-    // goes through each word in the file
-    std::string word;
-    while (file >> word) {
-        const int minWordLength{ 5 };
+  file.close();
 
-        word = cleanUpWord(word);
+  return true;
+}
 
-        // only words of length at least 5 are added to the map
-        if (word.length() >= minWordLength) {
-            ++wordCounter[word];
-        }
-    }
+/** Converts a word to lowercase and removes punctuation
+ *
+ * @param word The word to be cleaned up
+ * @return The cleaned up word
+ *
+ */
+std::string cleanUpWord(std::string &word) {
+  std::string newWord{};
+  std::string punctuation{".,?'\"!():"};
```

Comment:
This is a compile time constant, so I'd have this variable in captials too (and make it `const`!)